### ISSUE TITLE
(too) many ways to get the number of hits

### ISSUE THREAD

**Issue by JostMigenda** [WCSim/WCSim#175]
“[There are lots of ways to get the number of hits](https://github.com/WCSim/WCSim/blob/2f7ca11b89dc893f115932186184a94614e43ac2/sample-root-scripts/read_PMT.C#L48)” and even after removing the duplicate (PR will come shortly) we have several remaining:

1. `GetNumTubesHit()`
2. `GetNcherenkovhits()`
3. `GetCherenkovHits()->GetEntries()`

and

1. `GetNumDigiTubesHit()`
2. `GetNcherenkovdigihits()`
3. `GetCherenkovDigiHits()->GetEntries()`

Can we add comments or documentation to make it clear when to use which of these three? From looking at the code, that is not obvious to me.

**Comment by tdealtry** [comment]
My understanding is that 2/3 are equivalent, showing the actual number of hits. 1 is different, showing the number of unique PMTs that have been hit.
Very simple example - PMT 0 hit twice. Method one gives answer as 1; methods two/three give the answer as 2.
But I think documentation improvements are never a bad idea

**Comment by JostMigenda** [comment]
That’s what I thought from the method names, too. But when I run a simple test (50 MeV electron at origin) and analyse the result with `read_PMT.C`, all three methods return the same number; which is not what I expected.

To add to my confusion, I just discovered that in `doc/DetectorDocumentation.pdf` on p. 18, both `GetNumTubesHit()` and `GetNcherenkovhits()` are described as “number of tubes with a true hit”.

**Comment by tdealtry** [comment]
Looking at `WCSimEventAction::FillRootEvent()`

Hits
* `GetNumTubesHit()` filled with `WCDC_hits->entries()` - the number of hit phototubes
* `GetNcherenkovhits()` incremented with a loop over `WCDC_hits` - will also be the number of hit phototubes
* `GetNcherenkovhittimes()` incremented with a loop over `WCDC_hits`, and a loop over that PMTs' hit times - will be the total number of hits

Digits
* `GetNumDigiTubesHit()` - incremented with a loop over `WCDC`, and a loop over that PMTs' digits within that gate - will be the total number of digits in the gate
* `GetNcherenkovdigihits()` - incremented with a loop over `WCDC`, and a loop over that PMTs' digits within that gate - will be the total number of digits in the gate

(The `GetEntries()` methods are trivial alternatives to the `GetNcherenkov*()` convenience methods)

The names are confusing, so would suggest
* Make `GetNumDigiTubesHit()` do what the method name says
* Possibly adding `InGate` to the name of the digit-related methods (but of course these are called from WCSimRootTrigger, which is already split into 'gates', so not strictly required)
* Possibly remove `GetNcherenkovhits()`? And rename `GetNcherenkovhittimes()` -> `GetNcherenkovhits()`. This will require a lot of variable/method renaming in `WCSimRootEvent`...

What do people think?

**Comment by JostMigenda** [comment]
All these methods count both the number of (PMTs hit by) Cherenkov photons as well as the number of (PMTs with) dark noise events, right? So even having “Cherenkov” in the method names is a bit confusing, and I would suggest renaming `GetNcherenkovhittimes()` to `GetNumHits()`.

**Comment by JostMigenda** [comment]
Quick note: I’ll be at a summer school from 4th September and might be inclined to spend an evening or two renaming the variables while I’m there. @eosullivan , could you take a look at this before then and let us know what you think?

**Comment by eosullivan** [comment]
@JostMigenda, sure, I'll look this over and get back to you.

**Comment by eosullivan** [comment]
I agree with cleaning up this stuff in principle. A couple of comments:

* I don't like the idea of changing a function to return a different thing, since if someone is using this in their macro they might not realize that the value they are returning now has a different meaning. Deprecate GetNumDigiTubesHit() and create a newly named function that returns the number of PMT hits that were digitized.
* Let's make sure to document things better than the existing code. There should be inline explanations, a comment in the sample ROOT macros, and an update to the doc/DetectorDocumentation.pdf
* Finally, renaming things in the code seems simple, but mistakes can cause problems. Make sure you validate your changes where possible.

**Comment by tdealtry** [comment]
In terms of documentation, don't forget #99 - adding the comments with doxygen syntax now will be useful for the future
